homework4/homework4.js

```diff
@@ -0,0 +1,123 @@
+/* 1. Создать функцию multiply, которая будет принимать любое количество чисел и возвращать
+их произведение: multiply(1,2,3) = 6 (1*2*3)
+Если нет ни одного аргумента, вернуть ноль: multiply() // 0*/
+let multiplay = function () {
+    let result = 1;
+    let a;
+    if (arguments[0] === undefined){
+      return 0;
+
+    } else {
+       for (i = 0; i < arguments.length; i++){
+           typeof arguments[i] === 'number'
+               ? result *= arguments[i]
+               : result = 'error'
+        }
+       return result;
+    }
+};
+
+console.log('1 try = ' + multiplay());
+console.log('2 try = ' + multiplay(1,2,4,1));
+console.log('3 try = ' + multiplay('dflkdfkd', 'sdsdsdsd'));
+
+/*2.Создать функцию, которая принимает строку и возвращает строку-перевертыш:
+reverseString(‘test’) // “tset”.*/
+let shiftWord = function(shifter){
+let normalWord = '';
+for (i = 1; i < shifter.length + 1; i++) {
+    normalWord = normalWord + shifter[shifter.length - i]
+}
+    return normalWord};
+
+console.log(shiftWord('tset'));
+
+/*3. Создать функцию, которая в качестве аргумента принимает строку из букв и возвращает строку,
+где каждый символ разделен пробелом и заменен на юникод-значение символа:
+getCodeStringFromText(‘hello’) // “104 101 108 108 111”*/
+let getCodeStringFromText = function (word) {
+    let uniCode = '';
+   for (i =0; i < word.length;i++ ) {
+       uniCode += word.charCodeAt(i) + ' ';
+   }
+   return uniCode;
+}
+console.log(getCodeStringFromText('Hello'));
+/*4. Создать функцию угадай число. Она принимает число от 1-10 (обязательно проверить что
+число не больше 10 и не меньше 0). Генерирует рандомное число от 1-10 и сравнивает с
+переданным числом если они совпали то возвращает “Вы выиграли” если нет то “Вы не угадали
+ваше число 8 а выпало число 5”. Числа в строке указаны как пример вы подставляете реальные
+числа.*/
+let letsPlay = function (num) {
+    if (num > 0 && num < 11) {
+        let min = 1;
+        let max = 10;
+        var answer = '';
+        let randomNumber = Math.round(Math.random() * (max - min) + min);
+        (num === randomNumber) ?
+            answer = 'Вы выиграли' : answer = 'Вы не угадали ваше число ' + num + ' а выпало число ' + randomNumber;
+    } else {
+    answer = 'Загадайте число от 1 до 10';
+    }
+    return answer;
+}
+console.log(letsPlay(2) );
+
+/*5. Создать функцию, которая принимает число n и возвращает массив, заполненный числами от 1 до
+n: getArray(10); // [1,2,3,4,5,6,7,8,9,10]*/
+let getArray = function (max) {
+    let arr = [];
+    for (i= 0; i < max;i++){
+        arr[i] = i+1;
```

Comment:
You could just write `for(i = 1; i <= max; i++)` and don't do confusing thing - `i + 1`